Error after update from 5.2.0 to 5.3.0

```
error: expected expression, found `let` statement
  --> api/src/config/swagger.rs:54:9
   |
54 |         dto::page::Response<dto::get::unit::Response>,
   |         ^^^
   |
   = note: only supported directly in conditions of `if` and `while` expressions
   
error[E0061]: this method takes 2 arguments but 3 arguments were supplied
   --> api/src/config/swagger.rs:13:10
    |
13  | #[derive(OpenApi)]
    |          ^^^^^^^
...
54  |         dto::page::Response<dto::get::unit::Response>,
    |         --- unexpected argument #3
    |
note: method defined here
   --> /Users/artemiokost/.cargo/registry/src/index.crates.io-6f17d22bba15001f/utoipa-5.3.0/src/openapi/schema.rs:136:12
    |
136 |     pub fn schema<S: Into<String>, I: Into<RefOr<Schema>>>(mut self, name: S, schema: I) -> Self {
    |            ^^^^^^
    = note: this error originates in the derive macro `OpenApi` (in Nightly builds, run with -Z macro-backtrace for more info)
```
Error occurs when I'm using generic type
```
   components(schemas(
        dto::page::Response<dto::get::unit::Response>,
        dto::page::Response<dto::get::user::Response>,
    ))
```
    
Previously it was ok

--- a/utoipa-gen/CHANGELOG.md
+++ b/utoipa-gen/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Fixed
+
+* Fix bug in generic schemas on OpenApi derive macro (https://github.com/juhaku/utoipa/pull/1277)
+
 ### Changed
 
 * Update axum to v0.8 (https://github.com/juhaku/utoipa/pull/1269)

--- a/utoipa-gen/src/openapi.rs
+++ b/utoipa-gen/src/openapi.rs
@@ -633,7 +633,7 @@ impl Parse for Components {
     }
 }
 
-impl crate::ToTokensDiagnostics for Components {
+impl ToTokensDiagnostics for Components {
     fn to_tokens(&self, tokens: &mut TokenStream) -> Result<(), Diagnostics> {
         if self.schemas.is_empty() && self.responses.is_empty() {
             return Ok(());
@@ -659,7 +659,10 @@ impl crate::ToTokensDiagnostics for Components {
                         <#type_path as utoipa::ToSchema>::schemas(&mut schemas);
                         schemas
                     } )});
-                    components.extend(quote! { .schema(#name, #schema) });
+                    components.extend(quote! { .schema(#name, {
+                        let mut generics = Vec::<utoipa::openapi::RefOr<utoipa::openapi::schema::Schema>>::new();
+                        #schema
+                    }) });
 
                     components
                 },

--- a/utoipa-gen/tests/openapi_derive.rs
+++ b/utoipa-gen/tests/openapi_derive.rs
@@ -525,3 +525,51 @@ fn openapi_resolvle_recursive_references() {
 
     assert_json_snapshot!(schemas);
 }
+
+#[test]
+fn derive_generic_openapi_component_schemas() {
+    #[derive(Serialize, ToSchema)]
+    #[schema(as = dto::page::Response)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Response<T: Serialize> {
+        pub list: Vec<T>,
+        pub num: u64,
+        pub size: u64,
+        pub total: u64,
+    }
+
+    pub mod unit {
+        use serde::Serialize;
+        use utoipa::ToSchema;
+
+        #[derive(Serialize, ToSchema)]
+        #[schema(as = dto::get::unit::Response)]
+        #[serde(rename_all = "camelCase")]
+        pub struct Response {
+            pub id: i64,
+            pub latitude: f64,
+            pub longitude: f64,
+            pub title: String,
+            pub description: Option<String>,
+            pub country: String,
+            pub region: Option<String>,
+            pub city: String,
+            pub address: String,
+        }
+    }
+
+    #[derive(OpenApi)]
+    #[openapi(
+        components(
+            schemas(
+                Response<unit::Response>,
+            )
+        )
+    )]
+    struct ApiDoc;
+
+    let doc = serde_json::to_value(ApiDoc::openapi()).expect("OpenApi is JSON serializable");
+    let schemas = doc.pointer("/components");
+
+    assert_json_snapshot!(schemas)
+}

